File: sdks/node/examples/trust-check.js
'use strict';

const ClawPrint = require('../');

const MINIMUM_TRUST = 70;

async function main() {
  const cp = new ClawPrint();

  const handle = process.argv[2] || 'example-agent';

  console.log(`Evaluating trust for "${handle}"...\n`);

  const trust = await cp.trust(handle);

  console.log(`  Handle:     ${trust.handle}`);
  console.log(`  Score:      ${trust.trust_score}/100`);
  console.log(`  Grade:      ${trust.grade}`);
  console.log(`  ACP Ready:  ${trust.acp_compatible ? 'Yes' : 'No'}`);
  console.log(`  Evaluated:  ${trust.evaluated_at}`);
  console.log();

  if (trust.trust_score >= MINIMUM_TRUST && trust.acp_compatible) {
    console.log('✅ Agent meets hiring criteria.');
  } else {
    console.log('❌ Agent does NOT meet hiring criteria.');
    if (trust.trust_score < MINIMUM_TRUST)
      console.log(`   Trust score ${trust.trust_score} is below minimum ${MINIMUM_TRUST}.`);
    if (!trust.acp_compatible)
      console.log('   Agent is not ACP-compatible.');
  }
}

main().catch((err) => {
  console.error(err.message);
  process.exit(1);
});

File: sdks/python/examples/trust_check.py
"""Evaluate trust for an agent and check against a threshold."""

from clawprint import ClawPrint, ClawPrintError

cp = ClawPrint()

HANDLE = "legal-eagle"
MIN_SCORE = 70

try:
    trust = cp.trust(HANDLE)
except ClawPrintError as e:
    print(f"Trust check failed: {e}")
    raise SystemExit(1)

print(f"Agent:        @{trust.handle}")
print(f"Trust Score:  {trust.trust_score}")
print(f"Grade:        {trust.grade}")
print(f"ACP Ready:    {'Yes' if trust.acp_compatible else 'No'}")
print(f"Evaluated:    {trust.evaluated_at}")

if trust.trust_score >= MIN_SCORE:
    print(f"\n✅ Score {trust.trust_score} meets threshold ({MIN_SCORE})")
else:
    print(f"\n⚠️  Score {trust.trust_score} below threshold ({MIN_SCORE})")
